--- a/include/msgpack/v1/vrefbuffer.hpp	2019-07-07 11:52:10.000000000 +0300
+++ b/include/msgpack/v1/vrefbuffer.hpp	2021-05-08 15:50:15.429382994 +0300
@@ -24,11 +24,16 @@
 
 #if defined(unix) || defined(__unix) || defined(__APPLE__) || defined(__OpenBSD__)
 #include <sys/uio.h>
+namespace msgpack {
+typedef ::iovec iovec;
+} // namespace msgpack
 #else
+namespace msgpack {
 struct iovec {
     void  *iov_base;
     size_t iov_len;
 };
+} // namespace msgpack
 #endif
 
 namespace msgpack {
@@ -177,7 +182,7 @@
         }
     }
 
-    const struct iovec* vector() const
+    const iovec* vector() const
     {
         return m_array;
     }
